cc/mcts_player.cc

```diff
@@ -232,18 +232,40 @@ void MctsPlayer::TreeSearch() {
 
 void MctsPlayer::SelectLeaves(MctsNode* root, int num_leaves,
                               std::vector<MctsPlayer::TreePath>* paths) {
-  int max_iterations = num_leaves * 2;
+  DualNet::Output cached_output;
+
+  int max_cache_misses = num_leaves * 2;
   int num_selected = 0;
-  for (int i = 0; i < max_iterations; ++i) {
+  int num_cache_misses = 0;
+  int num_cache_hits = 0;
+  while (num_cache_misses < max_cache_misses) {
```

Comment:
I don't see how this condition can ever trigger

Reply:
It triggers near the end of the game where SelectLeaf would otherwise repeatedly call `IncorporateEndGameResult` because it keeps hitting `game_over` or `at_move_limit` leaves.